### PR TITLE
Enable browser cache for assets

### PR DESCRIPTION
**Pull request by nurali-techie**
This PR enables browser cache for all the resource under '_assets' directory for [fabric8-ui](https://github.com/fabric8-ui/fabric8-ui) application.

Fix - https://github.com/fabric8-ui/fabric8-ui-openshift-nginx/issues/18

### CODE REVIEW

**Comment by fabric8cd** [comment]
@nurali-techie snapshot fabric8-ui-openshift-nginx image is available for testing.  `docker pull fabric8/fabric8-openshift-nginx:SNAPSHOT-PR-17-1`

File: root/etc/nginx/nginx.conf [hunk 1]
```diff
@@ -60,65 +60,17 @@ http {
           etag off;
           if_modified_since off;
           more_clear_headers 'Last-Modified';
+          location =/index.html {
+            expires epoch;
```
Comment (nurali-techie) [inline root/etc/nginx/nginx.conf:64]: epoch will set cache expires to very long past (year 1970) which in effect works like expire always.

File: root/etc/nginx/nginx.conf [hunk 2]
```diff
@@ -60,65 +60,17 @@ http {
...
 
         location /_assets/ {
           etag on;
           gzip on;
           if_modified_since off;
           more_clear_headers 'Last-Modified';
-          # expires 1y;
-        }
-
-        # CORS proxy to OSO
```
Comment (nurali-techie) [inline root/etc/nginx/nginx.conf:73]: removed as no longer required as per @aslakknutsen

File: root/etc/nginx/nginx.conf [hunk 3]
```diff
@@ -60,65 +60,17 @@ http {
...
-            more_set_headers 'Access-Control-Allow-Methods: GET, POST, OPTIONS, PUT, DELETE';
-            more_set_headers 'Access-Control-Allow-Credentials: true';
-            more_set_headers 'Access-Control-Allow-Headers: Origin,Content-Type,Accept';
-            add_header Content-Length 0;
-            add_header Content-Type text/plain;
-            return 204;
-          }
-
-          proxy_pass      $oso;
+          expires 1y;
```
Comment (nurali-techie) [inline root/etc/nginx/nginx.conf:73]: this will set cache expiry for 1 year for all the content under _assets folder.

**Comment by nurali-techie** [comment]
**Test Cases:**

**Test Setup:**
- built f8-nginx docker image (with all changes).
- use f8-nginx image in f8-ui project and build f8-ui docker image.
- start f8-ui docker image.

**Tests:**
**1. query for resource under _assets**

Request: curl http://localhost:8080/_assets/images/badconnection323c5e4668736224da821f2c71070307.svg -I

Output:
HTTP/1.1 200 OK
Server: nginx/1.10.2
Date: Fri, 23 Mar 2018 09:31:45 GMT
Content-Type: image/svg+xml
Content-Length: 21869
Connection: keep-alive
ETag: "5ab4be64-556d"
**Expires**: Sat, 23 Mar **2019** 09:31:45 GMT
**Cache-Control**: max-age=31536000
Accept-Ranges: bytes

Verify: Check Cache-Control and Expires present with 1 yr expiry

**2. query for index.html**

Request: curl localhost:8080/index.html -I

Output:
HTTP/1.1 200 OK
Server: nginx/1.10.2
Date: Fri, 23 Mar 2018 12:52:00 GMT
Content-Type: text/html
Content-Length: 6840
Connection: keep-alive
Expires: Thu, 01 Jan **1970** 00:00:01 GMT
Cache-Control: **no-cache**
Accept-Ranges: bytes

Verify: Check Cache-Control set to 'no-cache'

**3. query for resource not under _assets**

Request: curl localhost:8080/_config/oauth.json -I

Output:
HTTP/1.1 200 OK
Server: nginx/1.10.2
Date: Fri, 23 Mar 2018 12:54:46 GMT
Content-Type: application/json
Content-Length: 486
Connection: keep-alive
Accept-Ranges: bytes

Verify: Check no cache related headers presents.

**4. change of resource name under _assets**

Steps:
1. login into osio - localhost:8080
2. make sure one of the resource (for ex - _assets/lib/polyfills.68242bc73dcace7ef76e.bundle.js) is cahced in browser cache.
3. chage /fabric8-ui/src/polyfills.browser.ts content. Just add one newline.
4. npm and docker build f8-ui and start docker image
5. login into osio - localhost:8080 (don't refresh)

Verify: Check name of "polyfills.68242bc73dcace7ef76e.bundle.js" is changed (hash part) and it's requested from server (and not from cached).

**Review by nurali-techie — COMMENTED** [review]
Required changes are done and tested.

**Review by aslakknutsen — APPROVED** [review]

**Comment by fabric8cd** [comment]
@nurali-techie snapshot fabric8-ui-openshift-nginx image is available for testing.  `docker pull fabric8/fabric8-openshift-nginx:SNAPSHOT-PR-17-2`

**Pull request merged**